iroh-dns-server/src/http/doh/extract.rs

```diff
@@ -98,53 +99,63 @@ pub struct DnsRequestQuery(pub(crate) DNSRequest, pub(crate) DnsMimeType);
 #[derive(Debug)]
 pub struct DnsRequestBody(pub(crate) DNSRequest);
 
-#[async_trait]
 impl<S> FromRequestParts<S> for DnsRequestQuery
 where
     S: Send + Sync,
 {
     type Rejection = AppError;
 
-    async fn from_request_parts(parts: &mut Parts, state: &S) -> Result<Self, Self::Rejection> {
-        let ConnectInfo(src_addr) = ConnectInfo::from_request_parts(parts, state).await?;
-
-        match parts.headers.get(header::ACCEPT) {
-            Some(content_type) if content_type == "application/dns-message" => {
-                handle_dns_message_query(parts, state, src_addr).await
-            }
-            Some(content_type) if content_type == "application/dns-json" => {
-                handle_dns_json_query(parts, state, src_addr).await
+    #[allow(clippy::manual_async_fn)]
+    fn from_request_parts(
+        parts: &mut Parts,
+        state: &S,
+    ) -> impl Future<Output = Result<Self, Self::Rejection>> + Send {
```

Comment:
I agree with clippy here, why the manual async fn?

Anyhow, seems like you've thought about it, given you've specifically added the `allow`, so I'm fine keeping it. Mostly curious & making sure it's not like an unintentionally left TODO.

Reply:
the return future seems to need an explicit send bound, and I could not get that to compile, feel free to try and make it so